core/src/test/java/io/parsingdata/metal/ErrorsTest.java

```diff
@@ -71,7 +73,17 @@ public void multiValueInRepN() throws IOException {
                 repn(dummy, ref("b"))
             );
        Optional<Environment> result = multiRepN.parse(stream(2, 2, 2, 2), enc());
-        assertFalse(result.isPresent());
+       assertFalse(result.isPresent());
+    }
+
+    @Test
+    public void environmentWithNegativeOffset() {
+        thrown.expect(IllegalArgumentException.class);
+        thrown.expectMessage("Argument offset may not be negative.");
+        new Environment(new ByteStream() {
+            @Override public byte[] read(BigInteger offset, int length) throws IOException { return new byte[0]; }
+            @Override public boolean isAvailable(BigInteger offset, int length) { return false; }
+        }, BigInteger.valueOf(-1));
```

Comment:
Why do we need a new (inline) `ByteStream` implementation here? Don't we have one handy inside the test scope already?

Reply:
Thing is, I don't want to pull in something like an `InMemoryByteStream` is all I want is a vanilla `ByteStream`. But maybe there's some other test version around...